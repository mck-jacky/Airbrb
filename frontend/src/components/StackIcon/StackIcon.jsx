import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '@mui/material';

const StackIcon = ({ ele1, ele2 }) => {
  return (
    <Box
      data-testid='stackicon'
      sx={{
        position: 'relative'
      }}
    >
      <span>
        {ele1}
      </span>
      <Box
        sx={{
          position: 'absolute',
          fontSize: 14,
          top: 0,
          left: 20
        }}
      >
        <span>
          {ele2}
        </span>
      </Box>
    </Box>
  );
};

StackIcon.propTypes = {
  ele1: PropTypes.node,
  ele2: PropTypes.node
}

export default StackIcon;
