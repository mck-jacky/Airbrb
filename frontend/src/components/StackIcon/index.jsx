import StackIcon from './StackIcon';

export default StackIcon
