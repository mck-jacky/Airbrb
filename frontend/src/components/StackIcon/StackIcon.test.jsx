import React from 'react';
import { render, screen } from '@testing-library/react';
import StackIcon from './StackIcon'
import ShallowRenderer from 'react-test-renderer/shallow'

describe('StackIcon', () => {
  it('is in the document', () => {
    render(<StackIcon ele1='Element1' ele2='Element2' />)
    const component = screen.getByTestId('stackicon')
    expect(component).toBeInTheDocument();
  })

  it('has element 1', () => {
    render(<StackIcon ele1='Element1' ele2='Element2' />)
    const component = screen.getByTestId('stackicon')
    expect(component).toContainHTML('<span>Element1</span');
  })

  it('has element 2', () => {
    render(<StackIcon ele1='Element1' ele2='Element2' />)
    const component = screen.getByTestId('stackicon')
    expect(component).toContainHTML('<span>Element2</span');
  })

  it('has to match the snapshot', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<StackIcon ele1='Element1' ele2='Element2' />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
