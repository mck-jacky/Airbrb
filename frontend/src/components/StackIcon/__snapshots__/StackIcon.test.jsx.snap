// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackIcon has to match the snapshot 1`] = `
<ForwardRef(Box)
  data-testid="stackicon"
  sx={
    Object {
      "position": "relative",
    }
  }
>
  <span>
    Element1
  </span>
  <ForwardRef(Box)
    sx={
      Object {
        "fontSize": 14,
        "left": 20,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <span>
      Element2
    </span>
  </ForwardRef(Box)>
</ForwardRef(Box)>
`;
